src/recovery/cubic.rs

```diff
@@ -173,22 +173,31 @@ fn on_packet_acked(
         }
     } else {
         // Congestion avoidance.
+        let mut ca_start_time = r.congestion_recovery_start_time;
 
-        // When we come here without congestion_event() triggered,
-        // This value can be None. In this case we initialize
-        // the value here.
-        if r.congestion_recovery_start_time.is_none() {
-            r.congestion_recovery_start_time = Some(now);
-
-            // This is also when the first congestion avoidance after a
-            // timeout. Following 4.7 of RFC, set k to 0 and reset
-            // w_max to cwnd during this period.
-            cubic.k = 0.0;
+        // In LSS, use LSS start time instead of congestion_recovery_start_time
+        // which can be None.
+        if r.hystart.enabled() &&
+            epoch == packet::EPOCH_APPLICATION &&
+            r.hystart.in_lss()
+        {
+            ca_start_time = r.hystart.lss_start_time();
 
+            // Reset w_max and k when LSS started.
+            if cubic.w_max == 0.0 {
+                cubic.w_max = r.congestion_window as f64;
+                cubic.k = 0.0;
+            }
+        } else if ca_start_time.is_none() {
+            // When we come here without congestion_event() triggered,
+            // initialize congestion_recovery_start_time, w_max and k.
+            ca_start_time = Some(now);
+            r.congestion_recovery_start_time = ca_start_time;
             cubic.w_max = r.congestion_window as f64;
+            cubic.k = 0.0;
         }
 
-        let t = now - r.congestion_recovery_start_time.unwrap();
+        let t = now - ca_start_time.unwrap();
```

Comment:
Compared to the old code, there seems to be a greater risk that the subtracting component is None.

Would this alternative work? 

* At L176, instead of declaring `ca_start_time` declare `let mut t = now;`
* Inside the LSS case, `if let Some(st) = r.hystart.lss_start_time() {t -= st};`
* In the else case,

```
else if r.congestion_recovery_start_time.is_none {
  r.congestion_recovery_start_time = Some(now);
  cubic.w_max = r.congestion_window as f64;
  cubic.k = 0.0;
  t -= now;
}

Reply:
> Inside the LSS case, if let Some(st) = r.hystart.lss_start_time() {t -= st};

L182 in_lss() = lss_start_time.is_some(). So inside if at L184, lss_start_time() is not None. So at the end of if (L200) ca_start_time is never None.

Reply:
@LPardue I tried to rewrite the logic a little cleaner to cover all the cases. PTAL